ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -138,9 +139,11 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
         if (quotaManager != null) {
           ThrottlingRecommendation throttlingRecommendation = quotaManager.getThrottleRecommendation(restRequest);
           if (throttlingRecommendation != null && throttlingRecommendation.shouldThrottle()) {
-            Map<String, String> quotaHeaderMap = RestUtils.buildUserQuotaHeadersMap(throttlingRecommendation);
-            throw new RestServiceException("User Quota Exceeded", RestServiceErrorCode.TooManyRequests, true, true,
+            if (quotaManager.getQuotaConfig().throttlingMode == QuotaMode.THROTTLING) {
```

Comment:
can we move this logic to `shouldThrottle` method in ThrottlingRecommendation? then we don't need two layers of if statement.

Reply:
I have changed the logic to remove the layer of if statement. We don't really need to separate `if`.